vault-plugins/flant_iam/backend/path_rolebindingapproval.go

```diff
@@ -0,0 +1,241 @@
+				"require_mfa": {
+					Type:        framework.TypeBool,
+					Description: "Necessity to approve second auth factor.",
+					Default:     false,
+				},
+				"require_unique_approver": {
+					Type:        framework.TypeBool,
+					Description: "Whether the approver is required to be unique among all approvals.",
+					Default:     true,
+				},
+			},
+			ExistenceCheck: b.handleExistence(),
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: b.handleUpdate(),
+					Summary:  "Update the role binding approval by ID.",
+				},
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: b.handleRead(),
+					Summary:  "Retrieve the role binding approval by ID.",
+				},
+				logical.DeleteOperation: &framework.PathOperation{
+					Callback: b.handleDelete(),
+					Summary:  "Deletes the role binding approval by ID.",
+				},
+			},
+		},
+	}
+}
+
+func (b *roleBindingApprovalBackend) handleExistence() framework.ExistenceFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (bool, error) {
+		tenantID := data.Get(model.TenantForeignPK).(string)
+		roleBindingID := data.Get(model.RoleBindingForeignPK).(string)
+
+		if !uuid.IsValid(roleBindingID) {
+			return false, fmt.Errorf("roleBindingID must be valid UUIDv4")
+		}
+
+		tx := b.storage.Txn(false)
+		repo := model.NewRoleBindingRepository(tx)
+
+		rb, err := repo.GetByID(roleBindingID)
+		if err != nil {
+			return false, err
+		}
+		exists := rb != nil && rb.TenantUUID == tenantID
+
+		return exists, nil
+	}
+}
+
+func (b *roleBindingApprovalBackend) handleUpdate() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		id := data.Get("uuid").(string)
+		if id == "" {
+			id = uuid.New()
+		}
+		userIDs := data.Get("users").([]string)
+		groupIDs := data.Get("groups").([]string)
+		serviceAccountIDs := data.Get("service_accounts").([]string)
+		requiredVotes := data.Get("required_votes").(int)
+		requireMFA := data.Get("require_mfa").(bool)
+		requireUniqueApprover := data.Get("require_unique_approver").(bool)
+
+		if requiredVotes <= 0 {
+			return nil, logical.CodedError(http.StatusBadRequest, "required_votes must be greater then zero")
+		}
+
+		roleBindingApproval := &model.RoleBindingApproval{
+			UUID:                  id,
+			TenantUUID:            data.Get(model.TenantForeignPK).(string),
+			RoleBindingUUID:       data.Get(model.RoleBindingForeignPK).(string),
+			Users:                 userIDs,
+			Groups:                groupIDs,
+			ServiceAccounts:       serviceAccountIDs,
+			RequiredVotes:         requiredVotes,
+			RequireMFA:            requireMFA,
+			RequireUniqueApprover: requireUniqueApprover,
+		}
+
+		tx := b.storage.Txn(true)
+		defer tx.Abort()
+
+		repo := model.NewRoleBindingApprovalRepository(tx)
+		_, err := repo.GetByID(id)
+		if err != nil {
```

Comment:
This decision making should be in the repo, not in handler. Handler is to parse input and form the response. All `if`'s should be in the repo.

Reply:
done